packages/jest-editor-support/src/SnapshotSupport.js

```diff
@@ -0,0 +1,151 @@
+    {
+      Identifier(node, state, parents) {
+        if (node.name === 'toMatchSnapshot') {
+          state.found.push({node, parents});
+        }
+      },
+    },
+    state
+  );
+
+  const position = state.found.findIndex(
+    node => node.node === toMatchSnapshot.node
+  ) + 1;
+
+  const fullName = parents.map(parent => parent.arguments[0].value).join(' ');
+
+  let describeLess = '';
+  if (parents[0].callee.name !== 'describe') {
+    // if `it` or `test` exists without a surrounding `describe`
+    // then `test ` is prepended to the snapshot fullName
+    describeLess = 'test ';
+  }
+
+  return `${describeLess}${fullName} ${position}`;
+};
+
+const snapshotPathFromFilePath = filePath => (
+  path.join(
+    path.dirname(filePath),
+    '__snapshots__',
+    path.basename(filePath) + '.snap'
+  )
+);
+
+const SnapshotSupport: SnapshotSupportType = {
+  getMetadataFromTestFile(filePath: string): Array<SnapshotMetadata> {
+    const fileContent = String(fs.readFileSync(filePath));
+    if (!SnapshotSupport.parser) {
+      throw new Error('SnapshotSupport.parser has to exists');
+    }
+    const fileNode = SnapshotSupport.parser(fileContent, {
+      plugins: SnapshotSupport.plugins,
+      sourceType: 'module',
+    });
+
+    const state = {
+      found: [],
+    };
+
+    walker(fileNode, {
+      Identifier(node, state, parents) {
+        if (node.name === 'toMatchSnapshot') {
+          state.found.push({node, parents});
+        }
+      },
+    }, state);
+
+    return state.found.map(toMatchSnapshot => {
+      const result = {
+        content: undefined,
+        exists: false,
+        exportName: 'N/A',
+        node: toMatchSnapshot.node,
+      };
+
+      const parents = toMatchSnapshot.parents.filter(
+        parent => (
+          parent.callee &&
+          ['test', 'it', 'describe'].indexOf(parent.callee.name) !== -1
+        )
+      );
+
+      const innerAssertion = parents[parents.length - 1];
+
+      if (!innerAssertion || innerAssertion.callee.name === 'describe') {
+        // dead code.
+        return result;
+      }
+
+      result.exportName = buildExportName(toMatchSnapshot, parents);
+      const snapshotPath = snapshotPathFromFilePath(filePath);
+      if (fs.existsSync(snapshotPath)) {
+        // reading and evaluating the snapshot
+        // instead of just requiring to avoid filling node cache
+        // eslint-disable-next-line no-new-func
+        const populate = new Function(
+          'exports',
+          String(fs.readFileSync(snapshotPath))
+        );
+
+        const snapshots = {};
+        populate(snapshots);
+        if (snapshots[result.exportName]) {
+          result.exists = true;
+          result.content = snapshots[result.exportName];
+        }
+      }
+
+      return result;
+    });
+  },
+};
+
+SnapshotSupport.plugins = [
+  'jsx', 'flow', 'objectRestSpread', 'classProperties',
+];
+SnapshotSupport.parser = babylon.parse.bind(babylon);
```

Comment:
Can we re-use code from here: https://github.com/cpojer/jest/blob/master/packages/jest-editor-support/src/parsers/BabylonParser.js#L46 ? We shouldn't make assumptions about the syntax that the user is using.

cc @orta